### PR TITLE
[Merged by Bors] - used separate mutex for addrbook locaAddresses

### LINKED ISSUE #2007: TODO: lock local for updates
Starting working on `TODO: lock local for updates`, part of #32 

## Description
Will use separate `RWMutext` for `localAddresses`. 

## Affected code

This issue appears in commit hash: 
685c99d0d8e224ab38fd9dc955f6d92b3cca9040

Related files: 
https://github.com/spacemeshos/go-spacemesh/blob/685c99d0d8e224ab38fd9dc955f6d92b3cca9040/p2p/discovery/addrbook.go#L94

Mutex will be used in: 
https://github.com/spacemeshos/go-spacemesh/blob/685c99d0d8e224ab38fd9dc955f6d92b3cca9040/p2p/discovery/addrbook.go#L108-L112

https://github.com/spacemeshos/go-spacemesh/blob/685c99d0d8e224ab38fd9dc955f6d92b3cca9040/p2p/discovery/addrbook.go#L115-L120

### PR DESCRIPTION
**Pull request by anton-kuklin**
## Motivation
#2007 
Closes #2007 

## Changes
Used separate RWMutex for `addrBook` `localAddresses` 

## Test Plan
Used existing package unit tests.

### CODE REVIEW

File: p2p/discovery/addrbook.go [hunk 1]
```diff
@@ -106,16 +107,16 @@ type addrBook struct {
...
 func (a *addrBook) AddLocalAddress(addr *node.Info) {
-	a.mtx.Lock()
+	a.localAddrMtx.Lock()
 	a.localAddresses = append(a.localAddresses, addr)
-	a.mtx.Unlock()
+	a.localAddrMtx.Unlock()
 }
 
 // IsLocalAddress returns true if this address was added as a local address before.
 func (a *addrBook) IsLocalAddress(addr *node.Info) bool {
```
Comment (y0sher) [inline p2p/discovery/addrbook.go:116]: since you changed mutexs, every occurrence of `isLocalAddressUnlocked` should be now using `IsLocalAddress`

**Review by y0sher — CHANGES_REQUESTED** [review]
Thanks for your contribution! added a comment :)

File: p2p/discovery/addrbook.go (hunk 1 above)
Comment (anton-kuklin) [reply p2p/discovery/addrbook.go:116]: yep, you are right, my fault

**Review by y0sher — APPROVED** [review]
Good job!
bors merge

**Comment by anton-kuklin** [comment]
@y0sher truth to be told, I got no idea where the bug is, that 'bors' failed. May it be an error in integration tests or somewhat?

**Comment by y0sher** [comment]
@justplesh not your fault. we're having some issues with the testing infra we'll merge as soon as it will resolve.

**Comment by anton-kuklin** [comment]
@y0sher is there any way I can help to resolve an issue with infra?

**Comment by y0sher** [comment]
it's an elasticsearch logs delay, we're moving to a new and improved testing infra that won't have these issues. thanks for the good will!
I refreshed the indices, lets try again :) 
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by noamnelke** [comment]
bors merge

**Comment by y0sher** [comment]
@noamnelke , looks like you have an automated script for that lol..
maybe will be better to add more timeout in the tests ?

**Comment by noamnelke** [comment]
haha, I'm lazy, but you're probably right...

**Comment by noamnelke** [comment]
see, no need 😝

**Comment by anton-kuklin** [comment]
@noamnelke wait, there will be more PR's soon ))

**Pull request closed without merging**